cdap-watchdog/src/main/java/co/cask/cdap/metrics/runtime/MetricsTwillRunnable.java

```diff
@@ -119,7 +125,14 @@ static Injector createGuiceInjector(CConfiguration cConf, Configuration hConf) {
       new MetricsStoreModule(),
       new AuditModule().getDistributedModules(),
       new AuthorizationEnforcementModule().getDistributedModules(),
-      new AuthenticationContextModules().getMasterModule()
+      new AuthenticationContextModules().getMasterModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(OwnerAdmin.class).to(DefaultOwnerAdmin.class);
+          bind(UGIProvider.class).to(RemoteUGIProvider.class).in(Scopes.SINGLETON);
```

Comment:
I don't think we need to bind UGIProvider here since we don't to any impersonation at all in metrics. Does the Guice injection unit test fails ?